src/main/java/com/epam/aidial/core/limiter/RateLimiter.java

```diff
@@ -84,26 +79,21 @@ public Future<RateLimitResult> limit(ProxyContext context) {
         }
     }
 
-    public Future<LimitStats> getLimitStats(String deploymentName, ProxyContext context) {
+    public Future<LimitStats> getLimitStats(Deployment deployment, ProxyContext context) {
         try {
             // skip checking limits if redis is not available
             if (resourceService == null) {
                 return Future.succeededFuture();
             }
             Key key = context.getKey();
-            Limit limit;
-            if (key == null) {
-                limit = getLimitByUser(context, deploymentName);
-            } else {
-                limit = getLimitByApiKey(context, deploymentName);
-            }
+            Limit limit = getLimitByUser(context, deployment);
             if (limit == null) {
                 log.warn("Limit is not found. Trace: {}. Span: {}. Key: {}. User sub: {}. Deployment: {}",
                         context.getTraceId(), context.getSpanId(), key == null ? null : key.getProject(),
-                        context.getUserSub(), deploymentName);
+                        context.getUserSub(), deployment.getName());
                 return Future.succeededFuture();
             }
-            return vertx.executeBlocking(() -> getLimitStats(context, limit, deploymentName), false);
+            return vertx.executeBlocking(() -> getLimitStats(context, limit, deployment.getName()), false);
         } catch (Throwable e) {
             return Future.failedFuture(e);
```

Comment:
Is it really used?

Reply:
Yes, it's. The controller LimitController calls this method to obtain stats.